tests/checks/gettext.fish

```diff
@@ -0,0 +1,27 @@
+# RUN: fish=%fish %fish %s
+#REQUIRES: command -v msgfmt
+
+set -l custom_gettext_var asdf
+
+set -lx LANG de_DE.utf8
+
+# Check if translations work.
+
+$fish --not-a-real-flag
+# CHECKERR: fish: --not-a-real-flag: unbekannte Option
+
+set --show custom_gettext_var
+# CHECK: $custom_gettext_var: Gesetzt in local Gültigkeit als nicht exportiert, mit 1 Elementen
+# CHECK: $custom_gettext_var[1]: |asdf|
```

Comment:
complex translations like this are certainly flawed --
"als" makes only sense if "pathvarv" is given, it should be a comma here.

	    let pathvarv = if var.is_pathvar() {
	        wgettext!(" a path variable")
	    } else {
	        L!("")
	    };
	    let vals = var.as_list();
	    streams.out.append(wgettext_fmt!(
	        "$%ls: set in %ls scope, %ls,%ls with %d elements",
	        var_name,
	        scope_name,
	        exportv,
	        pathvarv,
	        vals.len()
	    ));